app/Export/GetImages.hs

```diff
@@ -1,25 +1,161 @@
 import qualified Data.Map as M
 import System.Random
 import Svg.Generator
 import Export.ImageConversion
 import Happstack.Server (Request, rqCookies, cookieValue)
 import Data.List.Utils (replace)
+import Data.List.Split (splitOn)
+import Database.CourseQueries (getLectureTime, getTutorialTime)
+import Database.Tables as Tables
+import Control.Monad.IO.Class  (liftIO)
+import Data.List (partition)
+import Database.Persist.Sqlite (runSqlite, runMigration)
+import Config (databasePath)
+import Data.Fixed
 
 -- | If there is an active graph available, an image of that graph is created,
 -- otherwise the Computer Science graph is created as a default.
 -- Either way, the resulting graph's .svg and .png names are returned.
 getActiveGraphImage :: Request -> IO (String, String)
 getActiveGraphImage req = do
-    let cookies = M.fromList $ rqCookies req
+    let cookies = M.fromList $ rqCookies req  --  Map String Cookie
         graphName =
             replace "-" " " $
-                maybe "Computer-Science" cookieValue (M.lookup "active-graph" cookies)
+                maybe "Computer-Science" cookieValue (M.lookup "active-graph" cookies)  -- if M.lookup "active-graph" cookies is Nothing, then "Computer Science", else get cookieValue of coookie
     getGraphImage graphName (M.map cookieValue cookies)
 
+
+-- =================================
+
+-- data CourseInfo = CourseInfo {code :: String, section :: String, session :: String, time :: [Time]} deriving (Show)
+
+-- | Get lectures selected by user from selected-lectures cookie. If DNE, assume no lecture seleted
+-- get "CSC148H1-L0301-S_STA257H1-T0105-F_STA257H1-L0101-F_STA303H1-L0101-S_CSC148H1-T5101-F"
+-- return [[Time {timeField = [2.0,18.0]},Time {timeField = [2.0,18.5]},Time {timeField = [2.0,19.0]},Time {timeField = [2.0,19.5]},Time {timeField = [2.0,20.0]},Time {timeField = [2.0,20.5]}],[Time {timeField = [0.0,14.0]},Time {timeField = [0.0,14.5]},Time {timeField = [0.0,15.0]},Time {timeField = [0.0,15.5]},Time {timeField = [2.0,14.0]},Time {timeField = [2.0,14.5]}],Just [Time {timeField = [2.0,10.0]},Time {timeField = [2.0,10.5]},Time {timeField = [0.0,10.0]},Time {timeField = [0.0,10.5]},Time {timeField = [4.0,10.0]},Time {timeField = [4.0,10.5]}], [Time {timeField = [4.0,11.0]},Time {timeField = [4.0,11.5]},Time {timeField = [4.0,12.0]},Time {timeField = [4.0,12.5]}]]
+-- [["8:00","","","","",""],["9:00","","","","",""],["10:00","CSC108 (L)","","CSC108 (L)","","CSC108 (L)"],["11:00","","","","",""],["12:00","","","","",""],["1:00","","","","",""],["2:00","STA355 (L)","","STA355 (L)","",""],["3:00","STA355 (L)","","","",""],["4:00","","","","",""],["5:00","","","","",""],["6:00","","","","",""],["7:00","","","","",""],["8:00","","","","",""]]
+getActiveTimetable :: Request -> String -> IO (String, String)
+getActiveTimetable req session = do
+    -- get cookie value of "selected-lectures" from browser
+    let cookies = M.fromList $ rqCookies req  --  Map String Cookie
```

Comment:
This type comment would be more appropriately written as `let cookies :: Map String Cookie = ...`